Add varlink

fossdd opened the pull request:
## Description

https://varlink.org/

Adds varlink grammar and language

[systemd](https://github.com/systemd/systemd) started using it (see https://media.ccc.de/v/all-systems-go-2024-276-varlink-now-), so I think it's getting popular


## Checklist:
- [x] **I am adding a new language.**
  - [ ] The extension of the new language is used in hundreds of repositories on GitHub.com.
    - Search results for each extension:
      <!-- Replace FOOBAR with the new extension, and KEYWORDS with keywords unique to the language. Repeat for each extension added. -->
      -  https://github.com/search?type=code&q=NOT+is%3Afork+path%3A*.varlink
  - [x] I have included a real-world usage sample for all extensions added in this PR:
    - Sample source(s):
      - https://github.com/varlink/com.redhat.system/blob/master/src/com.redhat.system.varlink
    - Sample license(s):
      - MIT and Apache
  - [x] I have included a syntax highlighting grammar: https://github.com/varlink/language-varlink
      <!-- Setting a color is strongly recommended, but optional: `#cccccc` is used by default -->
  - [x] I have added a color
    - Hex value: `#42f1f4`
    - Rationale: It's the prefered color of varlink
  - [x] I have updated the heuristics to distinguish my language from others using the same extension.

lildude commented:
> systemd started using it (see https://media.ccc.de/v/all-systems-go-2024-276-varlink-now-), so I think it's getting popular

With only 51 indexed files, it's got a very long way to go to meet our usage requirements.

lildude commented on vendor/README.md:
```diff
@@ -597,9 +597,14 @@ This is a list of grammars that Linguist selects to provide syntax highlighting
+=======
+- **Vala:** [technosophos/Vala-TMBundle](https://github.com/technosophos/Vala-TMBundle)
+- **varlink:** [varlink/language-varlink](https://github.com/varlink/language-varlink)
+>>>>>>> Add varlink grammar
```
You need to fix this

fossdd replied:
Oh yeah, thanks fixed it!

lildude reviewed (changes requested):
Looks like you've still got remnants of a merge conflict that need cleaning up.

lildude commented on lib/linguist/languages.yml:
```diff
@@ -8590,6 +8590,17 @@ sed:
   type: markup
   color: "#66D0DD"
+varlink:
+  type: programming
```
Is this really a programming language. Feels more like configuration so would fall under data

fossdd replied:
I think it's a markup language as a varlink file describes interfaces (see https://en.wikipedia.org/wiki/Interface_description_language). What do you think?

lildude replied:
Markup is for things that result in visual effects, think html. From what I've read, the only varlink files that will be committed are the interface definition files which are data by the fact they're definition files… they're defining the methods the interface implements.

fossdd commented:
Thanks for the review! I addressed the issues. Yeah, varlink still isn't used that much (even though it's a few years old now), but I think with systemd adopting varlink it has a pretty good starting point.

lildude reviewed (changes requested):
The cached licence file needs to be the one generated by the `script/add-grammar` script you should have used to add the grammar. If you did not use this script, revert your changes and use it and then push your changes. Don't worry about keeping the history clean as we squash all commits when merging.
